Drivers/IWDG_driver/Src/IndependentWatchdog.cpp

```diff
@@ -0,0 +1,45 @@
+#include "IndependentWatchdog.h";
+#include "main.h"
+
+/**
+ * @brief Borrow the configuration of an already existing watchdog and set it
+ * to the prescaler and Reload values.
+ *
+ */
+IWDG_t::IWDG_t(IWDG_HandleTypeDef *watchdog) {
+	IWDG_InitTypeDef initDef = watchdog->Init;
+	this->iwdg_prescaler = initDef.Prescaler;
+	this->iwdg_reload = initDef.Reload;
+}
+
+/**
+ * @brief Initalize the driver using custom IWDG_Prescaler and realod values.
+ *
+ */
+IWDG_t::IWDG_t(uint8_t iwdg_prescaler, uint16_t iwdg_reload) {
+  // assert(IS_IWDG_PRESCALER(iwdg_prescaler)) Should this be added since it is needed when reinitialzing the watchdog
+  // assert(IS_IWDG_RELOAD(iwdg_reload))
+	this->iwdg_prescaler = iwdg_prescaler;
+	this->iwdg_reload = iwdg_reload;
+}
+
+/**
+ * @brief Reset the reload and counter values inside a watchdog
+ * using the member variables of IWDG_t
+ *
+ */
+HAL_StatusTypeDef IWDG_t::InitWatchdog(IWDG_HandleTypeDef *watchdog) {
```

Comment:
Is it possible to make this function return a bool? Want to abstract away the HAL stuff.

Reply:
Hi, I modified it to return  a bool, (refresh==HAL_OK)